### ISSUE TITLE
Allow turbofish in `view!` macro to better support generic components

### ISSUE THREAD

**Issue by tyoeer** [leptos-rs/leptos#2127]
**Is your feature request related to a problem? Please describe.**
I have a generic component whose input is a `MaybeSignal` of the generic. In order to avoid having to manually wrap the parameters every time, I added `#prop(into)`:
```rust
#[component]
fn Component<Object>(
	#[prop(into)]
	object: MaybeSignal<Object>,
) -> impl IntoView {}
```
However, when trying to actually pass a signal (`RwSignal<MyObject>`) to this, it can no longer infer the type due to overlapping implementations of `From`/`Into`:
```
error[E0283]: type annotations needed
    -snip-
    = note: multiple `impl`s satisfying `leptos::MaybeSignal<_>: std::convert::From<leptos::RwSignal<MyObject>>` found in the `leptos_reactive` crate:
            - impl<T> std::convert::From<T> for leptos::MaybeSignal<T>;
            - impl<T> std::convert::From<leptos::RwSignal<T>> for leptos::MaybeSignal<T>;
    = note: required for `leptos::RwSignal<MyObject>` to implement `std::convert::Into<leptos::MaybeSignal<_>>`
```

**Describe the solution you'd like**
The ability to use standard turbofish syntax to specify the type of the component, so Rust can infer the type automatically. E.g.:
```rust
let object = create_rw_signal(my_object);
view! {
	<Component::<MyObject> object />
}
```

**Describe alternatives you've considered**
Adding an optional ZST prop to the component that can be used to help type inference. This is rather clunky though:
```rust
#[component]
fn Component<Object>(
	#[prop(into)]
	object: MaybeSignal<Object>,
	#[prop(optional)]
	_type: [Object; 0],
) -> impl IntoView {}
```
```rust
let object = create_rw_signal(my_object);
let type_spec: [MyObject; 0] = [];
view! {
	<Component::<MyObject> object _type = type_spec />
}
```

**Additional context**
My WIP project where I ran into this:
- The generic component: https://github.com/tyoeer/fusta-femas/blob/f0d3f097e9a706b0c735d2f741d1c2991a3b4c80/app/src/table.rs#L189-L202
- Where the type can't be inferred: https://github.com/tyoeer/fusta-femas/blob/f0d3f097e9a706b0c735d2f741d1c2991a3b4c80/app/src/entry/details.rs#L114-L122
  - Changing the `entry::Model` on line 116 to `_` will trigger the error again

**Comment by gbj** [comment]
The view macro already supports generic syntax `<Component<T>/>`. In your case I think this is
```
use entry::Model;
// ...

<table::ObjectFieldValueList<Model> object=entry/>
```
I think this is what you mean, right? Let me know if not.

**Comment by tyoeer** [comment]
That does work, thanks!

I wasn't expecting that though, because:
- [The book just says that specifying generics on components (with turbofish) isn't supported](https://book.leptos.dev/view/03_components.html#optional-generic-props)
- I thought components are fancy functions (you can directly use them for the view prop of a route), and thus need generics specified as a function and not as types


Slightly related note: it doesn't support `::`, it errors on the second `:` with "expected identifier"

**Comment by gbj** [comment]
Yeah the book is just wrong, sorry. I'll update the docs, which wasn't done when this feature was added.

> I thought components are fancy functions (you can directly use them for the view prop of a route), and thus need generics specified as a function and not as types

> Slightly related note: it doesn't support `::`, it errors on the second `:` with "expected identifier"

Components *are* just fancy functions, but the `view` macro is also just pure syntax sugar. I would assume that the decision to use `<Component<T>/>` rather than `<Component::<T>/>` and the inability to parse `<Component<foo::T>/>` are both just artifacts of the fact that it's quite hard to parse a stream of arbitrary Rust tokens as valid (X)HTML.

**Comment by gbj** [comment]
Ok, updated docs here https://github.com/leptos-rs/book/commit/550428351ec6daa09e8da9ec517d8c8893cee23a
